codec/address/hex_codec.go

```diff
@@ -0,0 +1,51 @@
+package address
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"cosmossdk.io/core/address"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type HexCodec struct {
+	HexPrefix string
```

Comment:
I've seen how this struct is used and there's something I don't fully understand. The prefixes passed at the callsites are "cosmos", "cosmosvaloper"... these are bech32 prefixes. My (correct?) understanding of Hex addresses is that those prefixes don't exist. So, should we do away with the prefix field here? Or are you keeping it unused to minimize changes with upstream (doesn't seem so as this file is added in the PR)?

Reply:
+1, in this case there's no need for the prefix if just using hex encoded addresses

Reply:
Yeah the original idea was to minimize changes and especially get your view on this topic.
So I have two choices:
1. Delete that `HexPrefix` and adapt the `HexCodec` usage in the codebase (currently that's the case in `heimdall`)
2. Keep the prefix and use `0x` by default

Reply:
Done (approach 2) in latest commit

Reply:
Resolved